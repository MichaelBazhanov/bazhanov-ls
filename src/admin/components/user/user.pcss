@import "../../../styles/mixins.pcss";
.user-components {
	display: flex;
	align-items: center;
}
.username {
	font-weight: 600;
	font-size: 18px;
	color: $text-color;

	@include desktopHd {
	}
	@include desktop {
	}
	@include tablets {
		font-size: 16px;
	}
	@include phones {
	}
}
.content {
	margin-left: 20px;
}
.btns {
	.btn {
		cursor: pointer;
		color: $text-color;
		padding: 0;
		margin: 0;
		font-size: 14px;
		line-height: 19px;
		text-decoration-line: underline;
		display: none;

		@include desktopHd {
		}
		@include desktop {
		}
		@include tablets {
			display: block;
		}
		@include phones {
		}
	}
}