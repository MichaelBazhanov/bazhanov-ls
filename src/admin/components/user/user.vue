<template>
	<div class="user-components">
		<avatar size="2.7" :src="userPic" />
		<div class="content">
			<div class="username">Бажанов Михаил</div>
			<div class="btns">
				<button type="button" class="btn" @click="$emit('logout', 'logout')">Выйти</button>
			</div>
		</div>
	</div>
</template>

<script>
import avatar from "../avatar";
export default {
	components: {
		avatar
	},
	computed: {
		userPic() {
			return require("../../../images/content/user.jpg").default
		}
	}
}
</script>

<style lang="postcss" scoped src='./user.pcss'>

</style>>
