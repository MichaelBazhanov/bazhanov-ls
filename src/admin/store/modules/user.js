export default {
	namespaced: true,
	state: {
		user: {}
	},
	mutations: {
		SET_USER: (state, user) => state.user = user,
		CLEAR_USER: state => state.user = {},
	},
	getters: {
		userIsLoggedIn: state => {
			const userObj = state.user; //сохраняем user из state
			const userObjIsEmpty = Object.keys(userObj).length === 0 && userObj.constructor === Object;
			//удостоверимся что у этого объекта нет никаких полей И функция которая создала этот объект была базовая функция конструктора Object

			return userObjIsEmpty === false //проверка если объект не пустой то user залогинен
		},
		userId: (state, getters) => {
			return getters.userIsLoggedIn ? state.user.id : '376'
		}
	},
	actions: {
		login({commit}, user) {
			commit('SET_USER', user)
		},
		logout({commit}) {
			localStorage.removeItem('token'); //что бы выйти удаляем токен
			location.reload(); //перезагружаем страничку для того что бы выкинуло на /login
		}
	},
}