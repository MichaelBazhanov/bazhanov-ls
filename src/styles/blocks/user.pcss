.user {
	display: flex;
	flex-direction: column;
	align-items: center;
}
.user__avatar-style{
	width: 200px;
	height: 200px;
	border-radius: 50%;
	overflow: hidden;
	margin: 0 80px 48px 80px;
	flex-shrink: 0;
}
.user__data {
	width: 100%;
	font-weight: 600;
}
