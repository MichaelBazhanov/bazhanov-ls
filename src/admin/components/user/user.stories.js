import user from "./user.vue";


export default {
	title: 'user',
	components: { user }
}

export const defaultView = () => ({
	template: `
		<user/>
	`,
	components: { user },
})

defaultView.story = {
	name: 'Стандартный вид',
}