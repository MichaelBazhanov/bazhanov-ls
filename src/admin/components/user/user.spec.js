import user from './user'
import { shallowMount } from "@vue/test-utils"

it('user snapshot + emitted()', async () => {
	//steps
	const wrapper = shallowMount(user)

	//actions
	await wrapper.find('.btn').trigger('click');

	//check
	await wrapper.vm.$nextTick(); //ожидаем асинхронного обновления всего
	expect(wrapper.emitted('logout')[0]).toEqual(['logout'])
	expect(wrapper.element).toMatchSnapshot()
})