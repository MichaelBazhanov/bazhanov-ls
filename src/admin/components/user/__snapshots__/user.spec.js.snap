// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user snapshot + emitted() 1`] = `
<div
  class="user-components"
>
  <avatar-stub
    size="2.7"
    src="https://picsum.photos/300/300"
  />
   
  <div
    class="content"
  >
    <div
      class="username"
    >
      Бажанов Михаил
    </div>
     
    <div
      class="btns"
    >
      <button
        class="btn"
        type="button"
      >
        Выйти
      </button>
    </div>
  </div>
</div>
`;
